x-pack/legacy/plugins/infra/server/routes/snapshot/types.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { InfraWrappableRequest } from '../../lib/adapters/framework';
+
+export interface InfraSnapshotNode {
+  path: InfraSnapshotNodePath[];
+
+  metric: InfraSnapshotNodeMetric;
+}
+
+export interface InfraSnapshotNodePath {
+  value: string;
+
+  label: string;
+
+  ip?: string | null;
+}
+
+export interface InfraSnapshotNodeMetric {
+  name: InfraSnapshotMetricType;
+
+  value?: number | null;
+
+  avg?: number | null;
+
+  max?: number | null;
+}
+
+export enum InfraSnapshotMetricType {
```

Comment:
We should use `SnapshotMetricTypeRT` from `common/inventory_models/types.ts` for this.